File: src/main/java/com/deepak/grocery_delivery/service/CartService.java
package com.deepak.grocery_delivery.service;

import com.deepak.grocery_delivery.dto.cart.AddToCartRequest;
import com.deepak.grocery_delivery.dto.cart.CartResponse;
import com.deepak.grocery_delivery.dto.cart.UpdateCartItemRequest;

public interface CartService {

    CartResponse addToCart(
            String email,
            AddToCartRequest request
    );

    CartResponse getCart(String email);

    CartResponse updateCartItem(
            String email,
            Long cartItemId,
            UpdateCartItemRequest request
    );

    void removeCartItem(
            String email,
            Long cartItemId
    );

    void clearCart(String email);
}

File: frontend/src/services/cartService.js
import api from "./api";

export const addToCart = async (productId, quantity) => {
    const response = await api.post("/api/cart/items", {
        productId,
        quantity
    });

    return response.data;
};

export const getCart = async () => {
    const response = await api.get("/api/cart");

    return response.data;
};

export const updateCartItem = async (cartItemId, quantity) => {
    const response = await api.put(
        `/api/cart/items/${cartItemId}`,
        {
            quantity
        }
    );

    return response.data;
};

export const removeCartItem = async (cartItemId) => {
    await api.delete(`/api/cart/items/${cartItemId}`);
};

export const clearCart = async () => {
    await api.delete("/api/cart");
};